src/wp-includes/rest-api.php

```diff
@@ -56,6 +57,11 @@ function register_rest_route( $namespace, $route, $args = array(), $override = f
 		);
 	}
 
+	if ( isset( $args['permissions_callback'] ) ) {
```

Comment:
I think we should probably also handle when an array of routes is passed to `register_rest_route`.